File: next-main/src/main/java/com/next/reservations/core/repository/ReservationRepository.java
package com.next.reservations.core.repository;

import com.next.reservations.core.domain.Reservation;
import com.next.reservations.core.domain.ReservationStatus;
import org.springframework.data.jpa.repository.JpaRepository;

import java.sql.Date;
import java.time.LocalDate;
import java.util.List;

public interface ReservationRepository extends JpaRepository<Reservation, Long> {

    List<Reservation> findAllByStatus(ReservationStatus status);

    List<Reservation> findAllByStatusInAndReservationDate(List<ReservationStatus> status, LocalDate date);
}

File: next-main/src/main/kotlin/com/next/reservations/core/repository/ReservationRepository.kt
package com.next.reservations.core.repository

import com.next.reservations.core.domain.Reservation
import com.next.reservations.core.domain.ReservationStatus
import org.springframework.data.jpa.repository.JpaRepository
import java.time.LocalDate

interface ReservationRepository : JpaRepository<Reservation, Long> {
    fun findAllByStatus(status: ReservationStatus): List<Reservation>
    fun findAllByStatusInAndReservationDate(status: List<ReservationStatus>, date: LocalDate): List<Reservation>

    fun findAllByStatusInAndReservationTimeIdInAndReservationDateAfter(status: List<ReservationStatus>,
                                                                       reservationDateIds: List<Long>,
                                                                       reservationDate: LocalDate): List<Reservation>

    fun findAllByStatusAndReservationDateAfter(status: ReservationStatus, date: LocalDate): List<Reservation>

    fun findAllByStatusAndPreviousReservationDateAfter(status: ReservationStatus, date: LocalDate): List<Reservation>
}
